multiqc/modules/snpsplit/snpsplit.py

```diff
@@ -0,0 +1,202 @@
+        )
+
+        self.snpsplit_data = dict()
+
+        # Parse log files generated with newer versions of SNPsplit
+        parsed_logs = []
+        for f in self.find_log_files('snpsplit/new'):
+            parsed_logs.append(self.parse_new_snpsplit_log(f))
+
+        # Parse log files generated with older versions of SNPsplit
+        for f in self.find_log_files('snpsplit/old', filehandles=True):
+            parsed_logs.append(self.parse_old_snpsplit_log(f))
+
+        # Go through all results and save if we got something
+        for parsed in parsed_logs:
+            if parsed:
+                s_name = self.clean_s_name(parsed[0], f['root'])
+                if s_name in self.snpsplit_data:
+                    log.warning("Replacing duplicate sample {}".format(s_name))
+                self.snpsplit_data[s_name] = parsed[1]
+                self.add_data_source(f, s_name=s_name)
+
+        if len(self.snpsplit_data) == 0:
+            raise UserWarning
+        log.info("Found {} reports".format(len(self.snpsplit_data)))
+
+        self.write_data_file(self.snpsplit_data, 'multiqc_snpsplit')
+
+        self.allele_tagging_section()
+        self.allele_sorting_section()
+
+    def parse_new_snpsplit_log(self, f):
+        data = next(yaml.load_all(f['f'], Loader=yaml.SafeLoader))
+        flat_data = {}
+        for k in data:
+            for sk in data[k]:
+                newkey = '{}_{}'.format(k.lower(), sk)
+                flat_data[newkey] = data[k][sk]
+        input_fn = data['Meta']['infile']
+        return [input_fn, flat_data]
+
+    def parse_old_snpsplit_log(self, f):
+        input_fn = None
+        d = dict()
+
+        for line in f['f'].splitlines():
+            # Parse the sample name
+            match = re.match(r"Input file:\W+'([^']+)'", line)
+            if match:
+                # Report can have two sections with two input files,
+                # the second sorting the output from the tagging
+                # Just take the first instance, which is the initial input file
+                if input_fn is None:
+                    input_fn = match.group(1)
+                continue
+
+            regex_patterns = [
+                # Allele-tagging report
+                ['tagging_skipped', r"Reads were unaligned and hence skipped: (\d+)"],
+                ['tagging_unassignable', r"(\d+) reads were unassignable"],
+                ['tagging_genome1', r"(\d+) reads were specific for genome 1"],
+                ['tagging_genome2', r"(\d+) reads were specific for genome 2"],
+                ['tagging_unassignableCT', r"(\d+) reads that were unassignable contained C>T SNPs"],
+                ['tagging_ambiguous', r"(\d+) reads did not contain one of the expected bases"],
+                ['tagging_conflictingSNPs', r"(\d+) contained conflicting allele-specific SNPs"],
```

Comment:
@felixkrueger -  I'm trying to match these old categories with the ones in the new YAML. So far I have:

* `tagging_skipped` - `tagging_unaligned`
* `tagging_unassignable` - `tagging_unassignable`
* `tagging_genome1` - `tagging_g1`
* `tagging_genome2` - `tagging_g2`
* `tagging_unassignableCT` - `tagging_CT_positions_skipped`
* `tagging_ambiguous` - **?**
* `tagging_conflictingSNPs` - **??**

Could you help with the last two please?

There are also loads of new categories:

* `tagging_N_containing_reads`
* `tagging_N_deletion`
* `tagging_N_not_known`
* `tagging_N_was_known_SNP`
* `tagging_SNP_annotation`
* `tagging_SNPs_stored`
* `tagging_bizarre`
* `tagging_multi_N_deletion`
* `tagging_no_snp`
* `tagging_non_N_containing_reads`
* `tagging_total_reads`
* `tagging_unassigned_but_ct`

Should I plot all of these as well? Do you have any documentation for these? The docs don't seem to mention them and there hasn't been a release for a long time 😉 

If you could write one-line descriptions for each category that would be great, and I'll add them to the help text for the plot 👍

Reply:
The reason I ask is that if I can save the fields from the old report style using the same keys as in the new reports, then all of the plots etc should work the same for both outputs.

Reply:
Ignoring most of these new categories for now. Managed to match up most things making use of a handy crib-sheet from @felixkrueger.